### PR TITLE
status page

### PR DESCRIPTION
**Pull request by akisblack**

### CODE REVIEW

File: src/routes/instances/Component.svelte [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	export let projectWebsite: string;
+	export let status: number;
+</script>
+
+<div class="status-wrapper">
+	<CardInner
+		title={name}
+		{description}
+		{icon}
+		position={status === 200 ? "Operational" : "Down"}
```
Comment (Odyssey346) [inline src/routes/instances/Component.svelte:17]: Maybe s/Operational/Up

File: svelte.config.js [hunk 2]
```diff
@@ -17,7 +17,9 @@ const config = {
 	],
 
 	kit: {
-		adapter: adapter(),
+		adapter: adapter({
+			edge: true
```
Comment (Odyssey346) [inline svelte.config.js:21]: Curious as to what this does

File: package.json [hunk 3]
```diff
@@ -14,8 +14,9 @@
 	"devDependencies": {
 		"@iconify-json/fa6-solid": "^1.1.7",
 		"@iconify-json/simple-icons": "^1.1.28",
-		"@sveltejs/adapter-static": "1.0.0-next.43",
+		"@sveltejs/adapter-node": "1.0.0-next.98",
 		"@sveltejs/kit": "1.0.0-next.504",
+		"axios": "^1.1.2",
```
Comment (Odyssey346) [inline package.json:19]: Unused dependency

**Review by Odyssey346 — CHANGES_REQUESTED** [review]
I have some suggestions

File: src/routes/instances/+page.server.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		nitter: nitter.status || nitter.code,
+		element: element.status || element.code,
+		piped: piped.status || piped.code,
+		searxng: searxng.status || searxng.code,
+		gitea: gitea.status || gitea.code,
+		portainer: portainer.status || portainer.code,
+		mailcow: mailcow.status || mailcow.code,
+		plausible: plausible.status || plausible.code
+	};
+};
```
Comment (Midou36O) [inline src/routes/instances/+page.server.ts:52]: Honestly i prefer having something that could not literally fetch every website status before returning, and no soleil levant won't be able to load this fast, You should also consider the situation where a website is not even respondiing, say a connection timeout.

File: src/routes/instances/+page.svelte [hunk 5]
```diff
@@ -1,73 +1,120 @@
...
+	/>
+	<Component
+		name="Plausible"
+		description="Analytics for our website."
+		icon="https://avatars.githubusercontent.com/u/54802774"
+		website="https://analytics.projectsegfau.lt/projectsegfau.lt"
+		projectWebsite="https://plausible.io/"
+		status={data.plausible}
+	/>
+</div>
```
Comment (Midou36O) [inline src/routes/instances/+page.svelte:111]: Dropping json support in favor of components? Seriously?

File: src/routes/instances/+page.server.ts (hunk 4 above)
Comment (Midou36O) [reply src/routes/instances/+page.server.ts:52]: The problem is that when i tested it there was 0 indication whatsoever that it was loading, i thought the button or link broke, so i clicked multiple times, which itself causes more unnecessary requests. Worst case scenario add a loading button ? (And noJS users don't even get the status page report). Another approach is to load the page and then ping every service and THEN output the result (again noJS users won't benefit from it but it's the price to pay for fast page loading.)

**Review by Midou36O — CHANGES_REQUESTED** [review]
It just feels wrong honestly.

File: src/routes/instances/+page.svelte (hunk 5 above)
Comment (Odyssey346) [reply src/routes/instances/+page.svelte]: It's fine, it's basically the same.

**Review by Odyssey346 — APPROVED** [review]
Looks good. Using components is fine. Also, I'd still like the "Loading status..." thing. Can you do that?

**Comment by Midou36O** [comment]
image

**Comment by Midou36O** [comment]
btw, instead of up & down you could use a circle like this ● (or emoji)

**Comment by GhostSlayer** [comment]
Drivet's bootleg

**Comment by Odyssey346** [comment]
> btw, instead of up & down you could use a circle like this ● (or emoji)

I agree, but keep the text and add a little "pulse" animation like this
vivaldi_e7RNxhpnXM

**Comment by GhostSlayer** [comment]
> > btw, instead of up & down you could use a circle like this ● (or emoji)
> 
> I agree, but keep the text and add a little "pulse" animation like this vivaldi_e7RNxhpnXM [ vivaldi_e7RNxhpnXM ](https://user-images.githubusercontent.com/57069715/195627747-e3d4b3b9-6dae-44f3-96ec-4cf26a77084a.gif) [ ](https://user-images.githubusercontent.com/57069715/195627747-e3d4b3b9-6dae-44f3-96ec-4cf26a77084a.gif)

Agree that looks cool

**Comment by akisblack** [comment]
> image

known bug, trying to fix

**Comment by Odyssey346** [comment]
bump

**Comment by Midou36O** [comment]
@akisblack

**Comment by Midou36O** [comment]
image

PR pushed back till the app directory in sveltekit is stable

**Comment by Odyssey346** [comment]
merge it anyways

**Comment by Odyssey346** [comment]
also he was talking about next

**Comment by Midou36O** [comment]
> merge it anyways

He didn't even integrate it with segfautils so no. The known bug is still not resolved

**Comment by Odyssey346** [comment]
ok but that is your job

**Comment by Midou36O** [comment]
> ok but that is your job

I didn't open the PR "\\\_(ツ)_/"

**Comment by Odyssey346** [comment]
WHAT

**Comment by Midou36O** [comment]
In favor of nextjs i presume?

**Pull request closed without merging**